Source/WebDriver/WebDriverService.cpp

```diff
@@ -407,11 +413,100 @@ bool WebDriverService::acceptHandshake(HTTPRequestHandler::Request&& request)
     return true;
 }
 
-void WebDriverService::handleMessage(WebSocketMessageHandler::Message&&, Function<void (WebSocketMessageHandler::Message&&)>&&)
+void WebDriverService::handleMessage(WebSocketMessageHandler::Message&& message, Function<void (WebSocketMessageHandler::Message&&)>&& completionHandler)
 {
     // https://w3c.github.io/webdriver-bidi/#handle-an-incoming-message
-    // TODO Implement message handling for actual BiDi commands
-    // https://bugs.webkit.org/show_bug.cgi?id=280501
+
+    if (!message.connection) {
+        WTFLogAlways("Incoming message without attached connection. Ignoring message.");
+        return;
```

Comment:
Makes sense.

FTR, in the case of this specific error, it'll still be an early return on the soup server, as the server won't know which connection to send this reply to, relying on the original message's connection. But at least the original handler will be unblocked.

In the following check right after this one, I'll update it to send InvalidSessionID instead of ignoring the `completionHandler` too.